### DISCUSSION TITLE
switch hdmi on/off on RPi4 with "vcgencmd" not working anymore

### DISCUSSION THREAD

**Discussion by Gifford47** [guysoft/FullPageOS#586] [Q&A] [answered]
Hi!

with fullpageos version 0.13.0 i could switch hdmi on and off with vcgencmd display_power 0/1. But with fullpageos version 0.14.0 (nightly) i can´t use vcgencmd anymore. I know that RPi4 on bookworm normally uses wayland but fullpageos still uses x11 as manager.
here`s the following output:

```
# vcgencmd display_power 0
display_power=1

# xset dpms force off
Authorization required, but no authorization protocol specified
xset:  unable to open display ":0.0"

#echo $XDG_SESSION_TYPE
(nothing/empty)

#printf 'Desktop: %s\nSession: %s\n' "$XDG_CURRENT_DESKTOP" "$GDMSESSION"
Desktop:
Session:
```

Does anyone know how i get back control of hdmi power? @guysoft

**Comment by apartmedia** [comment]
You using ssh for this? Is it working on a local terminal?
Did you try calls using with or without sudo?
When using ssh is the exported DISPLAY env var correct?

**Comment by Gifford47** [comment]
The result is the same as for local and ssh.
All command were issued as sudo.
```
pi@fullpageos:~ $ export DISPLAY=:0
pi@fullpageos:~ $ echo "$DISPLAY"
:0
pi@fullpageos:~ $ sudo su
root@fullpageos:/home/pi# export DISPLAY=:0
root@fullpageos:/home/pi# echo "$DISPLAY"
:0

```

**Comment by apartmedia** [comment, marked as answer]
Try not doing this as root…
My following test was successful.. 
Via SSH:
- login as user pi
- export display `export DISPLAY=:0`
- use xset to disable display `xset dpms force off`

that worked for me in FullPageOS latest nightly (0.14.0)…

**Reply by Gifford47** [reply]
Strange... For me it isn't working. I still get "Authorization required, but no authorization protocol specified".

**Reply by apartmedia** [reply]
I only get this same auth error if I try this command with sudo or user root.
The regular pi user is working fine…

**Reply by guysoft** [reply]
might be xhost permissions. There is a setting: ``xhost +` which you might need to set from within the pi user where it works. Its the one that started the x session.

**Reply by Gifford47** [reply]
Ok, thanks for the hints! I'll try and report it after my vacation

**Reply by Gifford47** [reply]
Thanks! With user pi it works as expected!
Only issue I have is, that the display goes into standby after some time without touch input. Can I change the screensaver somehow?

**Comment by apartmedia** [comment]
There are several xset commands available to manipulate power management, screensaver and screen blanking after idle time. To name a few…

```
@xset -dpms		#display power management system off 
@xset s noblank		#screen blanking off
 @xset s off		#screen saver off  

#optional stuff to play with
 @xset s 0 0		#timeout (inactivity, idle time), cycle (anti burn-in bg change time)
 @xset s noexpose	#no exposure events, preference to disable screen saver 
@xset dpms 0 0 0	#set DPMS timeouts to zero (Standby, Suspend, Off)
```

My favorite is using one  Xserver argument in lightdm.conf to keep the screen active all the time
`xserver-command=X -s 0 dpms`

**Reply by Gifford47** [reply]
Thanks, it works! 👍

**Comment by Gifford47** [comment]
One last question about pygame. I want to show a windows on top but it never shows up. Why?

```
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import pygame

pygame.init()
window = pygame.display.set_mode((400, 400))
font = pygame.font.SysFont(None, 100)
clock = pygame.time.Clock()
text = font.render("Text", True, (255, 255, 0))

bomb = pygame.image.load('tap.png').convert_alpha()
bomb_rect = bomb.get_rect(center = window.get_rect().center)
bomb_mask = pygame.mask.from_surface(bomb)
click_count = 0
click_text = font.render('touch: ' + str(click_count), True, "black")

run = True
while run:
    clock.tick(60)
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            run = False 
        if event.type == pygame.MOUSEBUTTONDOWN:
            image_x, image_y = event.pos[0] - bomb_rect.x, event.pos[1] - bomb_rect.y
            if bomb_rect.collidepoint(event.pos) and bomb_mask.get_at((image_x, image_y)):
                click_count += 1
                click_text = font.render('touch: ' + str(click_count), True, "black")

    window.fill((255, 255, 255))
    window.blit(bomb, bomb_rect)
    window.blit(click_text, click_text.get_rect(topleft = (20, 20)))
    pygame.display.flip()

pygame.quit()
exit()
```

**Reply by Gifford47** [reply]
The solution was to export the DISPLAY env var: ```export DISPLAY=:0``` before starting the script at first time
